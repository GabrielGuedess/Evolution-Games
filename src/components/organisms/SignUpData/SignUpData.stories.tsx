import { Story, Meta } from '@storybook/react/types-6-0';

import { SignUpData, SignUpDataProps } from './SignUpData';

const dataInputs = {
  name: '',
  lastName: '',
  cpf: '',
  cellphone: '',
  date: new Date(),
};

const setDataInputs = () => ({});
const setCurrentStep = () => ({});

export default {
  title: 'Organisms/SignUpData',
  component: SignUpData,
  args: {
    dataInputs,
    setDataInputs,
    setCurrentStep,
    currentStep: 1,
  },
} as Meta<SignUpDataProps>;

export const Default: Story<SignUpDataProps> = args => <SignUpData {...args} />;
