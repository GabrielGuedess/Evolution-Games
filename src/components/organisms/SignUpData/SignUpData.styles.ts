import media from 'styled-media-query';

import styled from 'styled-components';

export const Wrapper = styled.form`
  width: 100%;
  max-width: 40.4rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 3.2rem;

  > button span {
    padding: 0.9rem 0;
  }
`;

export const InputBox = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 3.2rem;

  ${media.greaterThan('medium')`
    gap: 1.6rem;
  `}
`;
