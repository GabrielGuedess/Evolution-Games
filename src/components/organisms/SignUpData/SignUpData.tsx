import { useState, Dispatch, SetStateAction } from 'react';
import InputMask from 'react-input-mask';

import { FieldErrors, signUpDataValidate } from 'utils/validations';

import Button from 'components/atoms/Button/Button';
import { InputForm } from 'components/atoms/InputForm/InputForm';

import * as S from './SignUpData.styles';

export type DataInputs = {
  name: string;
  lastName: string;
  cpf: string;
  cellphone: string;
  date: Date;
};

export type SignUpDataProps = {
  dataInputs: DataInputs;
  setDataInputs: Dispatch<SetStateAction<DataInputs>>;
  currentStep: number;
  setCurrentStep: (step: number) => void;
};

export const SignUpData = ({
  dataInputs,
  setDataInputs,
  currentStep,
  setCurrentStep,
}: SignUpDataProps) => {
  const [errorMessages, setErrorMessages] = useState<FieldErrors>({});

  const handleInput = (field: keyof DataInputs, value: string | Date) => {
    setDataInputs(s => ({ ...s, [field]: value }));

    const errors = signUpDataValidate({ ...dataInputs, [field]: value });

    setErrorMessages({ ...errorMessages, [field]: errors[field] });
  };

  const handleSubmit = async (event: React.FormEvent) => {
    event.preventDefault();

    if (
      dataInputs.date.toISOString().slice(0, 10) ===
      new Date().toISOString().slice(0, 10)
    ) {
      setErrorMessages({ date: 'Não pode ser hoje' });
      return;
    }

    if (
      Object.values(errorMessages).some(item => item !== undefined) ||
      Object.values(dataInputs).some(item => item === '')
    ) {
      setErrorMessages(errorMessages);
      return;
    }

    setErrorMessages({});
    setCurrentStep(currentStep + 1);
  };

  return (
    <S.Wrapper role="form" onSubmit={handleSubmit}>
      <S.InputBox>
        <InputForm
          required
          type="text"
          name="nome"
          label="Nome"
          labelFor="nome"
          value={dataInputs.name}
          placeholder="Insira seu nome"
          errorMessage={errorMessages?.name}
          onChange={e => handleInput('name', e.target.value)}
          isInvalid={!!errorMessages?.name}
        />

        <InputForm
          required
          type="text"
          name="sobrenome"
          label="Sobrenome"
          labelFor="sobrenome"
          value={dataInputs.lastName}
          placeholder="Insira seu sobrenome"
          errorMessage={errorMessages?.lastName}
          onChange={e => handleInput('lastName', e.target.value)}
          isInvalid={!!errorMessages?.lastName}
        />

        <InputForm
          required
          as={InputMask}
          mask="999.999.999-99"
          type="text"
          name="cpf"
          label="CPF"
          labelFor="cpf"
          value={dataInputs.cpf}
          placeholder="Insira seu CPF"
          errorMessage={errorMessages?.cpf}
          onChange={e =>
            handleInput('cpf', e.target.value.replace(/[^0-9]/g, ''))
          }
          isInvalid={!!errorMessages?.cpf}
        />

        <InputForm
          required
          type="text"
          as={InputMask}
          mask="(99) 99999-9999"
          name="celular"
          label="Celular"
          labelFor="celular"
          value={dataInputs.cellphone}
          placeholder="Insira seu celular"
          errorMessage={errorMessages?.cellphone}
          onChange={e =>
            handleInput('cellphone', e.target.value.replace(/[^0-9]/g, ''))
          }
          isInvalid={!!errorMessages?.cellphone}
        />

        <InputForm
          required
          type="date"
          name="date"
          label="Data de nascimento"
          labelFor="date"
          value={dataInputs.date.toISOString().slice(0, 10)}
          placeholder="Insira sua data de nascimento"
          errorMessage={errorMessages?.date}
          onChange={e => handleInput('date', new Date(e.target.value))}
          isInvalid={!!errorMessages?.date}
        />
      </S.InputBox>

      <Button
        disabled={
          Object.values(errorMessages).some(item => item !== undefined) ||
          Object.values(dataInputs).some(item => item === '')
        }
        as="button"
        type="submit"
        size="large"
        fullWidth
      >
        Próximo
      </Button>
    </S.Wrapper>
  );
};
