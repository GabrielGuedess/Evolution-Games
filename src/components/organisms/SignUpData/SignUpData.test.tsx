import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { SignUpData } from './SignUpData';

const dataInputs = {
  name: '',
  lastName: '',
  cpf: '',
  cellphone: '',
  date: '',
};

const setDataInputs = jest.fn();
const setCurrentStep = jest.fn();

describe('<SignUpData />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <SignUpData
        dataInputs={dataInputs}
        setDataInputs={setDataInputs}
        currentStep={0}
        setCurrentStep={setCurrentStep}
      />,
    );

    expect(
      screen.getByRole('heading', { name: /SignUpData/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
