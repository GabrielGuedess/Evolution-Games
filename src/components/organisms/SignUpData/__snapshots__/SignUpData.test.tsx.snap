// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<SignUpData /> should render the heading 1'] = `
.c7 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  position: relative;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 5px;
  -webkit-transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: #7C3AED;
  color: #F4F4F5;
  width: 100%;
}

.c7:not(:disabled):is(:hover,:focus) {
  box-shadow: 0 0.8rem 2.8rem rgba(118,74,241,0.5);
  color: #F4F4F5;
}

.c7 .effect {
  fill: #7C3AED;
}

.c7:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(30%);
  filter: saturate(30%);
}

.c7:not(:disabled):is(:hover,:focus) {
  -webkit-filter: brightness(1.1);
  filter: brightness(1.1);
}

.c7:not(:disabled):active {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c7 span {
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 1;
  display: block;
  position: relative;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  z-index: 1;
}

.c7:not(:disabled):is(:hover,:focus) span {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c7:not(:disabled):is(:hover,:focus) .icon {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c7:not(:disabled):active span {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c7 svg {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c7 .effect {
  position: absolute;
  top: initial;
  left: initial;
  width: 100%;
  height: 100%;
  z-index: -1;
}

.c7 .effect path {
  -webkit-transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c7:not(:disabled):is(:hover,:focus) .effect path {
  d: path( 'M0,0 C0,-5 100,-5 100,0 C105,0 105,100 100,100 C100,105 0,105 0,100 C-5,100 -5,0 0,0' );
}

.c7:not(:disabled):active .effect path {
  d: path( 'M0 5C0 2.23858 2.23858 0 5 0H95C97.7614 0 100 2.23858 100 5V95C100 97.7614 97.7614 100 95 100H5C2.23858 100 0 97.7614 0 95V5Z' );
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  position: relative;
}

.c6 {
  position: absolute;
  left: 1rem;
  top: 1.7rem;
  padding: 0 3px;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  z-index: 2;
}

.c6::before {
  content: '';
  position: absolute;
  inset: 0;
  width: 100%;
  z-index: -1;
}

.c4 {
  width: 100%;
  font-size: 1.8rem;
  font-weight: 400;
  font-style: normal;
  line-height: 3.2rem;
  border-width: 2px;
  border-style: solid;
  border-color: #272530;
  background-color: transparent;
  border-radius: 0.5rem;
  padding: 0.8rem;
  color: #FFFFFF;
  outline: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4::-webkit-calendar-picker-indicator {
  background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15" viewBox="0 0 24 24"><path fill="%23272530" d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"/></svg>');
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4 .c5 {
  color: #FFFFFF;
}

.c4:-webkit-autofill {
  -webkit-box-shadow: 0 0 0px 100rem #04040C inset;
  -webkit-text-fill-color: #FFFFFF;
}

.c4:-webkit-autofill ~ div,
.c4:-webkit-autofill:hover ~ div,
.c4:-webkit-autofill:focus ~ div,
.c4:-webkit-autofill:active ~ div {
  color: #717B8A;
}

.c4::-webkit-input-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4::-moz-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4:-ms-input-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4::placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c4:focus {
  border-color: #7C3AED;
}

.c4:focus::-webkit-calendar-picker-indicator {
  background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15" viewBox="0 0 24 24"><path fill="%237C3AED" d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"/></svg>');
}

.c4:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c4:focus::-moz-placeholder {
  opacity: 1;
}

.c4:focus:-ms-input-placeholder {
  opacity: 1;
}

.c4:focus::placeholder {
  opacity: 1;
}

.c4:is(:focus,:valid) + .c5 {
  top: -2rem;
  color: #7C3AED;
  font-size: 1.2rem;
}

.c4::selection {
  color: #FFFFFF;
  background: #7C3AED;
}

.c0 {
  width: 100%;
  max-width: 40.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3.2rem;
}

.c0 > button span {
  padding: 0.9rem 0;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3.2rem;
}

@media (min-width:450px) {
  .c7 .effect {
    top: -5%;
    left: -5%;
    width: 110%;
    height: 110%;
  }
}

@media (min-width:768px) {
  .c6::before {
    background: #04040C;
  }
}

@media (min-width:768px) {
  .c4:is(:focus,:valid) + .c5 {
    top: -8px;
  }
}

@media (min-width:768px) {
  .c1 {
    gap: 1.6rem;
  }
}

<form
  class="c0"
  role="form"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          autocomplete="on"
          class="c4"
          id="nome"
          name="nome"
          placeholder="Insira seu nome"
          required=""
          type="text"
          value=""
        />
        <label
          aria-label="nome"
          class="c5 c6"
          for="nome"
        >
          Nome
        </label>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          autocomplete="on"
          class="c4"
          id="sobrenome"
          name="sobrenome"
          placeholder="Insira seu sobrenome"
          required=""
          type="text"
          value=""
        />
        <label
          aria-label="sobrenome"
          class="c5 c6"
          for="sobrenome"
        >
          Sobrenome
        </label>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          autocomplete="on"
          class="c4"
          id="cpf"
          name="cpf"
          placeholder="Insira seu CPF"
          required=""
          type="text"
          value=""
        />
        <label
          aria-label="cpf"
          class="c5 c6"
          for="cpf"
        >
          CPF
        </label>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          autocomplete="on"
          class="c4"
          id="celular"
          name="celular"
          placeholder="Insira seu celular"
          required=""
          type="text"
          value=""
        />
        <label
          aria-label="celular"
          class="c5 c6"
          for="celular"
        >
          Celular
        </label>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          autocomplete="on"
          class="c4"
          id="date"
          name="date"
          placeholder="Insira sua data de nascimento"
          required=""
          type="date"
          value=""
        />
        <label
          aria-label="date"
          class="c5 c6"
          for="date"
        >
          Data de nascimento
        </label>
      </div>
    </div>
  </div>
  <button
    aria-label="Button Action"
    class="c7"
    disabled=""
    type="submit"
  >
    <span>
      Pr√≥ximo
    </span>
    <svg
      aria-hidden="true"
      class="effect"
      preserveAspectRatio="none"
      viewBox="-5 -5 110 110"
    >
      <path
        d="M0 8C0 3.58172 3.58172 0 8 0H92C96.4183 0 100 3.58172 100 8V92C100 96.4183 96.4183 100 92 100H8C3.58172 100 0 96.4183 0 92V8Z"
      />
    </svg>
  </button>
</form>
`;
